src/eda_server/api/job.py

```diff
@@ -91,6 +91,147 @@ async def create_job_instance(
     return {**j.dict(), "id": job_instance_id, "uuid": job_uuid}
 
 
+@router.post(
+    "/api/job_instance/{job_instance_id}",
+    response_model=schema.JobInstanceRead,
+    operation_id="rerun_job_instance",
+)
+async def rerun_job_instance(
+    job_instance_id: int, db: AsyncSession = Depends(get_db_session)
+):
+    query = sa.select(models.job_instances).where(
+        models.job_instances.c.id == job_instance_id
+    )
+    job_instance = (await db.execute(query)).first()
+    if not job_instance:
+        error = f"Job instance {job_instance_id} not found"
+        logger.error(error)
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail=error
+        )
+
+    logger.info("job_instance: %s", job_instance)
+
+    query = (
+        sa.select(models.activation_instances)
+        .select_from(models.activation_instances)
+        .join(
+            models.activation_instance_job_instances,
+            models.activation_instance_job_instances.c.job_instance_id
+            == job_instance.id,
+        )
+        .where(
+            models.activation_instance_job_instances.c.activation_instance_id
+            == models.activation_instances.c.id
+        )
+    )
+    activation_instance = (await db.execute(query)).first()
+
+    if not activation_instance:
+        error = f"Activation instance related with {job_instance_id} not found"
+        logger.error(error)
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail=error
+        )
+
+    logger.info("activation_instance: %s", activation_instance)
+
+    query = sa.select(models.playbooks).where(
+        models.playbooks.c.name == job_instance.name,
+        models.playbooks.c.project_id == activation_instance.project_id,
+    )
+    playbook_row = (await db.execute(query)).first()
+
+    if not playbook_row:
+        error = f"Playbook {job_instance.name} not found"
+        logger.error(error)
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail=error
+        )
+
+    logger.info("playbook_row: %s", playbook_row)
+
+    query = sa.select(models.extra_vars).where(
+        models.extra_vars.c.id == activation_instance.extra_var_id,
+    )
+    extra_var_row = (await db.execute(query)).first()
+
+    if not extra_var_row:
+        error = f"extra_var {activation_instance.extra_var_id} not found"
+        logger.error(error)
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail=error
```

Comment:
Should this be a 404 error instead of 400?

Reply:
I totally agree the first 400 error: `job_instance not found` should return error code 404. About the others, the missing are due to the broken relations. If we return 404, users may confuse because the job instance do exist.